Apps/RaptureIntTests/src/test/java/rapture/api/document/DocumentApiTest.java

```diff
@@ -0,0 +1,291 @@
+
+import org.testng.Assert;
+import org.testng.Reporter;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Optional;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+import rapture.common.RaptureScriptLanguage;
+import rapture.common.RaptureScriptPurpose;
+import rapture.common.RaptureURI;
+import rapture.common.Scheme;
+import rapture.common.client.HttpDocApi;
+import rapture.common.client.HttpEventApi;
+import rapture.nightly.IntegrationTestHelper;
+
+public class DocumentApiTest {
+    String raptureUrl = null;
+    private String raptureUser = null;
+    private String rapturePass = null;
+    private HttpLoginApi raptureLogin = null;
+    private HttpDocApi docApi=null;
+    private HttpEventApi eventApi=null;
+    private HttpScriptApi scriptApi=null;
+    IntegrationTestHelper helper=null;
+    
+    @BeforeClass(groups={"document","mongo", "nightly"})
+    @Parameters({"RaptureURL","RaptureUser","RapturePassword"})
+    public void beforeTest(@Optional("http://localhost:8665/rapture")String url, @Optional("rapture")String user, @Optional("rapture")String password)  {
+        raptureUrl=url;
+        raptureUser=user;
+        rapturePass=password;
+        raptureLogin = new HttpLoginApi(raptureUrl, new SimpleCredentialsProvider(raptureUser, rapturePass));
+        helper = new IntegrationTestHelper(raptureUrl, raptureUser, rapturePass);
+        try {
+            raptureLogin.login();
+            docApi = new HttpDocApi(raptureLogin);
+            scriptApi = new HttpScriptApi(raptureLogin);
+            eventApi = new HttpEventApi(raptureLogin);
+
+        } catch (RaptureException e) {
+            e.printStackTrace();
+        }   
+    }
+    
+    @Test(groups={"document","mongo", "nightly"})
+    public void testDocumentPut () {
+    	RaptureURI repo = helper.getRandomAuthority(Scheme.DOCUMENT);
+        helper.configureTestRepo(repo, "MONGODB");
+        String docPath= new RaptureURI.Builder(repo).docPath("doc"+System.nanoTime()).build().toString();
+        String content="{\"key1\":\"value1\"}";
+        Reporter.log("Creating and checking document "+docPath, true);
+        docApi.putDoc(docPath, content);
+        Assert.assertTrue(docApi.docExists(docPath));
+        Assert.assertEquals(docApi.getDoc(docPath),content);
+    }
+    
+    @Test(groups={"document","mongo", "nightly"})
+    public void testDocumentUpdate () {
+    	RaptureURI repo = helper.getRandomAuthority(Scheme.DOCUMENT);
+        helper.configureTestRepo(repo, "MONGODB");
+        String docPath= new RaptureURI.Builder(repo).docPath("doc"+System.nanoTime()).build().toString();
+        String content="{\"key1\":\"value1\"}";
+        Reporter.log("Creating and checking document "+docPath, true);
+        docApi.putDoc(docPath, content);
+        Assert.assertTrue(docApi.docExists(docPath));
+        Assert.assertEquals(docApi.getDoc(docPath),content);
```

Comment:
This first part is redundant - it's a duplicate of testDocumentPut above.
There's no point in repeating actions. 
The purpose of this suite is simply to act as verification that the build worked.
We don't need multiple test cases.